src/github.com/getlantern/go-loggly/loggly.go

```diff
@@ -0,0 +1,328 @@
+package loggly
+
+import . "github.com/visionmedia/go-debug"
+import . "encoding/json"
+import "io/ioutil"
+import "net/http"
+import "strings"
+import "bytes"
+import "time"
+import "sync"
+import "fmt"
+import "os"
+import "io"
+
+const Version = "0.4.3"
+
+const api = "https://logs-01.loggly.com/bulk/{token}"
+
+type Message map[string]interface{}
+
+var debug = Debug("loggly")
+
+var nl = []byte{'\n'}
+
+var (
+	defaultHTTPClient *http.Client
+	httpClientMu      sync.RWMutex
+)
+
+type Level int
+
+const (
+	DEBUG Level = iota
+	INFO
+	NOTICE
+	WARNING
+	ERROR
+	CRITICAL
+	ALERT
+	EMERGENCY
+)
+
+// Loggly client.
+type Client struct {
+	// Optionally output logs to the given writer.
+	Writer io.Writer
+
+	// Log level defaulting to INFO.
+	Level Level
+
+	// Size of buffer before flushing [100]
+	BufferSize int
+
+	// Flush interval regardless of size [5s]
+	FlushInterval time.Duration
+
+	// Loggly end-point.
+	Endpoint string
+
+	// Token string.
+	Token string
+
+	// Default properties.
+	Defaults Message
+	buffer   [][]byte
+	tags     []string
+	sync.Mutex
+}
+
+// SetHTTPClient sets the HTTP client the loggly client is going to use to
+// connect to the loggly API.
+func SetHTTPClient(newClient *http.Client) {
+	httpClientMu.Lock()
+	defaultHTTPClient = newClient
```

Comment:
Also, this should probably be a property on the loggly.Client instead of a global.